### DISCUSSION TITLE
Limit for the text that can be converted to speech

### DISCUSSION THREAD

**Discussion by BFRS02523** [SahilAggarwal2004/react-text-to-speech#4] [General]
Hi, I was trying to convert long text into speech, and the speech stops after some 100-150 characters. is there any limit set manually or some bug?

**Comment by SahilAggarwal2004** [comment]
Hey @BFRS02523,

Thanks for reaching out!

To answer your question, no, there is no manual limit set on the number of characters that are spoken. However, the browser may put some limit on the number of characters that can be spoken in its WebSpeech API implementation, which is usually much larger than 100-150 characters.

Can you show the exact text being passed and the browser and OS being used so that I can help you better?

**Reply by BFRS02523** [reply]
Hi Sahil, following is the text i am trying 
"
To cancel a return order, follow these steps:

Go to Returns from the left menu and select All to view return orders.
Search the Order ID or AWB to view complete details of your order. You can use the search bar present on top of the screen for this purpose.
Click on More options of that order and then click on the Cancel Order option from there.
A pop-up will appear for cancellation confirmation. Click on 'Yes, Cancel' if you are sure about cancellation.
Note: Cancellation functionality is not available in Out for Pickup status.

I hope this helps! Let me know if you have any further questions.
"
Browser is Chrome  124.0.6367.62 and os is mac

**Reply by SahilAggarwal2004** [reply]
Okay, I tried passing this text to the `<Speech>` component as below:
```typescript
import Speech from "react-text-to-speech";

export default function App() {
  return (
    <main>
      <div>react-text-to-speech</div>
      <Speech
        text=`To cancel a return order, follow these steps:

Go to Returns from the left menu and select All to view return orders.
Search the Order ID or AWB to view complete details of your order. You can use the search bar present on top of the screen for this purpose.
Click on More options of that order and then click on the Cancel Order option from there.
A pop-up will appear for cancellation confirmation. Click on 'Yes, Cancel' if you are sure about cancellation.
Note: Cancellation functionality is not available in Out for Pickup status.

I hope this helps! Let me know if you have any further questions.`
      />
    </main>
  );
}
```
And it's working absolutely fine and as expected. But I am on Windows and using Chrome 124.0.6367.119.

So this is probably an browser related issue with Chrome on macOS. Just to confirm, can you run the below code in your browser's console and check if it's working fine?
```javascript
const text = `To cancel a return order, follow these steps:

Go to Returns from the left menu and select All to view return orders.
Search the Order ID or AWB to view complete details of your order. You can use the search bar present on top of the screen for this purpose.
Click on More options of that order and then click on the Cancel Order option from there.
A pop-up will appear for cancellation confirmation. Click on 'Yes, Cancel' if you are sure about cancellation.
Note: Cancellation functionality is not available in Out for Pickup status.

I hope this helps! Let me know if you have any further questions.`;
const utterance = new SpeechSynthesisUtterance(text);
speechSynthesis.speak(utterance);
```

**Reply by SahilAggarwal2004** [reply]
Also, is there any way that you are unmounting the component? Because unmounting of the component will lead to cancellation of the speech.

**Reply by SahilAggarwal2004** [reply]
Hi @BFRS02523,

I apologize for the delayed response. While the issue was related to the `Web Speech API` rather than `react-text-to-speech`, I’ve addressed it by introducing the `maxChunkSize` prop in the `<Speech>` component in the latest version, v0.17.0.

You can use the updated component as follows, and this should help resolve the issue:
```ts
import Speech from "react-text-to-speech";

export default function App() {
  return (
    <main>
      <div>react-text-to-speech</div>
      <Speech
        text=`To cancel a return order, follow these steps:

Go to Returns from the left menu and select All to view return orders.
Search the Order ID or AWB to view complete details of your order. You can use the search bar present on top of the screen for this purpose.
Click on More options of that order and then click on the Cancel Order option from there.
A pop-up will appear for cancellation confirmation. Click on 'Yes, Cancel' if you are sure about cancellation.
Note: Cancellation functionality is not available in Out for Pickup status.

I hope this helps! Let me know if you have any further questions.`
        maxChunkSize={100}
      />
    </main>
  );
}
```

I hope this helps! Please let me know if you have any further questions.

**Comment by BFRS02523** [comment]
Thanks a lot.
Cheers

On Tue, Sep 10, 2024 at 1:17 PM Sahil Aggarwal ***@***.***>
wrote:

> Hi @BFRS02523 <https://github.com/BFRS02523>,
>
> I apologize for the delayed response. While the issue was related to the Web
> Speech API rather than react-text-to-speech, I’ve addressed it by
> introducing the maxChunkSize prop in the <Speech> component.
>
> You can use the updated component as follows, and this should help resolve
> the issue:
>
> import Speech from "react-text-to-speech";
> export default function App() {
>   return (
>     <main>
>       <div>react-text-to-speech</div>
>       <Speech
>         text="To cancel a return order, follow these steps:Go to Returns from the left menu and select All to view return orders.Search the Order ID or AWB to view complete details of your order. You can use the search bar present on top of the screen for this purpose.Click on More options of that order and then click on the Cancel Order option from there.A pop-up will appear for cancellation confirmation. Click on 'Yes, Cancel' if you are sure about cancellation.Note: Cancellation functionality is not available in Out for Pickup status.I hope this helps! Let me know if you have any further questions."
>         maxChunkSize={100}
>       />
>     </main>
>   );}
>
> I hope this helps! Please let me know if you have any further questions.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SahilAggarwal2004/react-text-to-speech/discussions/4#discussioncomment-10599227>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCRNI6RWRXFJ3TODNU55F6DZV2PZXAVCNFSM6AAAAABHKNQDJWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANJZHEZDENY>
> .
> You are receiving this because you were mentioned.Message ID:
> <SahilAggarwal2004/react-text-to-speech/repo-discussions/4/comments/10599227
> @github.com>
>

-- 
 <http://bit.ly/3JWf3ae>
